scripts/src/scverse_template_scripts/cruft_prs.py

```diff
@@ -0,0 +1,171 @@
+"""Script to send cruft update PRs.
+
+Uses `template-repos.yml` from `scverse/ecosystem-packages`.
+"""
+
+import os
+from collections.abc import Generator
+from dataclasses import dataclass, field
+from logging import basicConfig, getLogger
+from pathlib import Path
+from subprocess import run
+from tempfile import TemporaryDirectory
+from typing import IO, NotRequired, TypedDict, cast
+from urllib.parse import urlparse
+
+import typer
+from git.repo import Repo
+from git.util import Actor
+from github import Github
+from github.ContentFile import ContentFile
+from github.GitRelease import GitRelease as GHRelease
+from github.Repository import Repository as GHRepo
+from yaml import safe_load
+
+log = getLogger(__name__)
+
+
+@dataclass
+class GitHubConnection:
+    name: str
+    email: str
+    token: str = field(repr=False)
+    gh: Github = field(init=False)
+    sig: Actor = field(init=False)
+
+    def __post_init__(self) -> None:
+        self.gh = Github(self.token)
+        self.sig = Actor(self.name, self.email)
+
+    def auth(self, url_str: str) -> str:
+        url = urlparse(url_str)
+        # TODO: not mutable
+        url.username = self.name
+        url.password = self.token
+        return url
+
+
+@dataclass
+class PR:
+    release: GHRelease
+
+    @property
+    def title(self) -> str:
+        return f"Update template to {self.release.tag_name}"
+
+    @property
+    def branch(self) -> str:
+        return f"template-update-{self.release.tag_name}"
+
+    @property
+    def body(self) -> str:
+        template_usage = "https://cookiecutter-scverse-instance.readthedocs.io/en/latest/template_usage.html"
+        return f"""\
```

Comment:
since we insert the body, we need to `indent` that one. and once we use that, a refactor could break things as the `indent`, would need a different argument. I’d rather use a template file than that.

Reply:
ah got it!